src/models/experts/reducers.ts

```diff
@@ -1,19 +1,33 @@
 /* eslint-disable */
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
 import { LoadingStatusEnum } from '../../old/lib/types';
-import { IExpertsState } from './types';
-import { fetchExperts } from './asyncActions';
+import {
+  IExpertsState,
+  AutorsListOrder,
+  AutorsListSortBy,
+  ISortAutorsList,
+  ITextFields,
+} from './types';
+import { fetchExperts, fetchExpertsAutorsList } from './asyncActions';
 import { getAsyncActionsReducer } from '../helpers/asyncActions';
 
 const initialState: IExpertsState = {
   data: {
     expertIds: [],
     experts: {},
-    meta: {
-      totalPages: 0,
-      pageNumber: 0,
-      isLastPage: false,
-      totalElements: 0,
+    totalPages: 0,
+    isLastPage: true,
```

Comment:
yes, if the current page will be the last one of all existed pages

Reply:
then instead of saving extra flag we could just create separate selector:

```
isLastPageSelector = (pageNumber) => (state) => pageNumber === state.....totalPages - 1
```
and use it in component 

```
const currentPageNumber = useSelector ....
const isLastPage = useSelector(isLastPageSelector(currentPageNumber));
```

or 

```
isLastPageSelector = (state, page/number) => (state) => pageNumber === state.....totalPages - 1

```

with using as

```
const isLastPage = useSelector(state => isLastPageSelector(state, currentPageNumber));
```

what do you say?

Reply:
This also makes sense but in this case I should rewrite the logic in src/old/modules/experts/components/ExpertsView.tsx and src/old/modules/experts/components/ExpertMaterialsContainer.tsx as the component use the same slice of state as src/views/Profile/AuthorsList/AuthorsList.tsx and also remove isLastPage flag from the server side because the flag becomes unnecessary and there is no reason to get it in the response.

Reply:
decided not to do it this way, since `isLastPage` comes from server